src/libsync/creds/dummycredentials.cpp

```diff
@@ -27,6 +27,11 @@ QString DummyCredentials::user() const
     return _user;
 }
 
+QString DummyCredentials::password() const
+{
+    Q_ASSERT(false && "NOT IMPLEMENTED YET!");
```

Comment:
I'd use Q_UNREACHABLE() ('cause I don't think this can be ever called) and `return QString();`